src/main/java/io/jenkins/plugins/checks/github/GitHubChecksContext.java

```diff
@@ -1,15 +1,40 @@
 package io.jenkins.plugins.checks.github;
 
+import edu.hm.hafner.util.VisibleForTesting;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import hudson.model.Job;
+import hudson.model.Run;
+import jenkins.plugins.git.AbstractGitSCMSource;
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.SCMRevision;
+import org.jenkinsci.plugins.displayurlapi.DisplayURLProvider;
 import org.jenkinsci.plugins.github_branch_source.GitHubAppCredentials;
 import org.jenkinsci.plugins.github_branch_source.GitHubSCMSource;
-import hudson.model.Run;
-import jenkins.scm.api.SCMSource;
+import org.jenkinsci.plugins.github_branch_source.PullRequestSCMRevision;
 
+import java.util.Optional;
+
+/**
+ * Provides check properties that should be resolved  Jenkins job.
+ */
 class GitHubChecksContext {
-    private final GitHubSCMSource source;
-    private final GitHubAppCredentials credentials;
-    private final String headSha;
-    private final String url;
+    @Nullable
+    private final Run<?, ?> run;
+    private final Job<?, ?> job;
+    private final GitHubSCMFacade scmFacade;
+    private final DisplayURLProvider urlProvider;
+
+    /**
+     * Creates a {@link GitHubChecksContext} according to the run. All attributes are computed during this period.
+     *
+     * @param job
+     *         a run of a GitHub Branch Source project
+     */
+    @SuppressFBWarnings("NP_NONNULL_PARAM_VIOLATION")
```

Comment:
Why this suppression here and on the next constructor? If it is required there’s a justification parameter in the annotation you should use

Reply:
because pass null when calling method, I refactor the constructors just now.